### PR TITLE
add `workspaceAutoBackAndForth` option to i3 config

### PR DESCRIPTION
**Pull request by wedens**

### CODE REVIEW

File: modules/services/window-managers/i3.nix [hunk 1]
```diff
@@ -410,6 +410,15 @@ let
         '';
       };
 
+      workspaceAutoBackAndForth = mkOption {
+        type = types.bool;
+        default = false;
+        example = true;
+        description = ''
+          Assume you are on workspace "1: www" and switch to "2: IM" using mod+2 because somebody sent you a message. You don’t need to remember where you came from now, you can just press $mod+2 again to switch back to "1: www".
```
Comment (teto) [inline modules/services/window-managers/i3.nix]: this is one long line. @rycee shouldn't the linter fail on this one ?

Comment (wedens) [reply modules/services/window-managers/i3.nix]: I'm not sure how it's supposed to be formatted, but I've added some newlines 🙂

**Comment by rycee** [comment]
The commit message should be changed to

> i3: add `workspaceAutoBackAndForth` option

File: modules/services/window-managers/i3.nix (hunk 1 above)
Comment (rycee) [reply modules/services/window-managers/i3.nix]: @teto I'm not certain about how much the xml checks will look at line lengths. At the moment I'm more relying on the code reviewing for this 🙂

**Comment by wedens** [comment]
@rycee I've changed the message

**Comment by uvNikita** [comment]
Since we already have workspaceLayout option, it might make sense to create a new `workspace` namespace with two options: `workspace.layout`, and `workspace.autoBackAndForth`.

i3 options names are not perfect and quite often confusing in my opinion, and I think it would be nice to come up with a better name for this option too. So any ideas are welcome :) Although if we can't figure out something better, we can stick with the current name as well.

**Comment by teto** [comment]
I just was looking for this - indispensable - option and was surprised not to see it.

I am not too keen on adding yet a hierarichal level `workspace.` since it's a global setting anyway.

**Comment by uvNikita** [comment]
Rebased into master: https://github.com/rycee/home-manager/commit/5161dd3b2e9130d116809845187b088be2dc25b7. Thanks for the contribution!

**Pull request closed without merging**